### PR TITLE
Fix eye gaze simulation when XR isn't enabled

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

There's a fun quirk with the Unity input system when two types of the same name are registered. And unfortunately, both the Input System and Unity's OpenXR package _both_ register types of `PoseControl`. This causes one of the types (often the Input System's, since the OpenXR package has a bit of a delayed registration logic) to be overwritten by the other.

To work around this, we added an explicit re-registration of the Input System type with a _different_ identifying name, so the input system sees this as two separate types.

Unfortunately, something in Unity's rebinding logic sees these two controls as the same, even though the Input System itself sees them as unique. Therefore, when OpenXR registers its control while the simulated eye gaze is registered (which happens basically always in the editor, due to the delayed registration logic mentioned above), the Input System decides to clean up and recreate our simulated eye gaze device. This leads to our cached value becoming invalid and we need to query for the new device. Luckily, Unity maintains the device ID across these two devices.

Fixes https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10880

### CODE REVIEW

**Comment by Zee2** [comment]
What a spectacular headache. Thanks for working on this @keveleigh  :)

File: com.microsoft.mrtk.input/Simulation/InputSimulator.cs [hunk 1]
```diff
@@ -228,10 +223,7 @@ public EyeGazeSimulationSettings EyeGazeSettings
         /// </remarks>
         private void EnableSimulatedEyeGaze()
         {
-            if (simulatedEyeGaze == null)
-            {
-                simulatedEyeGaze = new SimulatedEyeGaze();
-            }
+            simulatedEyeGaze ??= new SimulatedEyeGaze();
```
Comment (Zee2) [inline com.microsoft.mrtk.input/Simulation/InputSimulator.cs:226]: Boy, wouldn't it be nice if we could use this syntax everywhere...

**Review by Zee2 — APPROVED** [review]

File: com.microsoft.mrtk.input/Simulation/InputSimulator.cs (hunk 1 above)
Comment (keveleigh) [reply com.microsoft.mrtk.input/Simulation/InputSimulator.cs:226]: It sure would!

**Review by davidkline-ms — APPROVED** [review]

**Pull request merged**